crates/sui-core/src/consensus_handler.rs

```diff
@@ -63,58 +63,74 @@ impl ConsensusHandler {
 impl ExecutionState for ConsensusHandler {
     /// This function will be called by Narwhal, after Narwhal sequenced this certificate.
     #[instrument(level = "trace", skip_all)]
-    async fn handle_consensus_transaction(
+    async fn handle_consensus_output(
         &self,
         // TODO [2533]: use this once integrating Narwhal reconfiguration
-        consensus_output: &Arc<ConsensusOutput>,
-        consensus_index: ExecutionIndices,
-        serialized_transaction: Vec<u8>,
+        consensus_output: ConsensusOutput,
     ) {
-        let index = Self::update_hash(
-            &self.last_seen,
-            consensus_index.clone(),
-            &serialized_transaction,
-        );
-        let index = if let Some(index) = index {
-            index
-        } else {
-            debug!(
+        let mut sequenced_transactions = Vec::new();
+        let mut seq = 0;
+
+        for (cert, batches) in consensus_output.batches {
```

Comment:
Note that `Notifier` had some metrics such as `notifier_processed_batches`, `batch_execution_latency` and `notifier_processed_bytes`. Unless I missed something it feels like we just abandoned them in this PR. Perhaps you can move them from narwhal to sui and populate them here?

Reply:
Ah good point there, since the notifier no longer does any processing I considered them obsolete. The meanings will be a bit different, so maybe we create equivalent ones with separate names. After the reordering of transactions, we won't be executing batches sequentially anymore for example.

Reply:
We can have different names, but I think measuring those things are still important.
The batch reordering will matter for `batch_execution_latency` metric only right? I think @akichidis added it, so perhaps he can add context on what should replace it with new architecture. Perhaps it can just measure time to when batch is deserialized in handler?